app/assets/stylesheets/govuk_publishing_components/components/govspeak/_typography.scss

```diff
@@ -15,6 +15,10 @@
 .gem-c-govspeak {
   @include markdown-typography;
 
+  &.gem-c-govspeak--font-size-l {
```

Comment:
I would structure this slightly differently (assuming there's no specificity problems) as

```Sass
.gem-c-govspeak {
  @include markdown-typography;
  // etc.
}

&.gem-c-govspeak--font-size-l {
  // etc.
}
```

This means that if we decided to add another size in future we don't have to keep overriding it - `gem-c-govspeak--font-size-l` is a modifier class.

Reply:
I'm just suggesting a slightly cleaner class structure. Our approach is to try to avoid nesting where possible, because that makes the tree more complicated and harder to change later (you have to keep overriding stuff).

It would be:

```Sass
.gem-c-govspeak {
  @include markdown-typography;
  // etc.
}

&.gem-c-govspeak--font-size-l {
  @include markdown-typography($base_font_size: 19, $font_size: 24, $heading_font_size: 36);
}
```

That would produce two separate classes, but we treat the second class as a modifier, so that when it's added to an element with the first class, it changes the styles. It's not a huge difference, it just avoids potential difficulties further down the road.

Reply:
Ah okay, gotcha.

I think we'd be relying on source order to break the specificity tie though - both rules just have single class selectors, so they're exactly as specific as each other. Is that correct? Is using source order something we're okay with?

If we specify both classes in the override, then it's definitely more specific.

Reply:
Just to be super clear, the markup in both cases is the same:

```
<div class="gem-c-govspeak gem-c-govspeak--font-size-l">...</div>
```